device/tt_soc_descriptor.h

```diff
@@ -162,16 +163,38 @@ class tt_SocDescriptor {
         ethernet_core_channel_map(other.ethernet_core_channel_map),
         trisc_sizes(other.trisc_sizes),
         device_descriptor_file_path(other.device_descriptor_file_path),
-        overlay_version(other.overlay_version),
-        unpacker_version(other.unpacker_version),
         dst_size_alignment(other.dst_size_alignment),
-        packer_version(other.packer_version),
         worker_l1_size(other.worker_l1_size),
         eth_l1_size(other.eth_l1_size),
         noc_translation_id_enabled(other.noc_translation_id_enabled),
-        dram_bank_size(other.dram_bank_size) {
+        dram_bank_size(other.dram_bank_size),
+        coordinate_manager(other.coordinate_manager) {
     }
-    private:
+    
+    // Coordinate conversions.
+
+    // Conversions from logical coordinates should be used just for worker cores.
+    tt_physical_coords logical_to_physical_coords(tt_logical_coords logical_coords);
+    tt_virtual_coords logical_to_virtual_coords(tt_logical_coords logical_coords);
+    tt_translated_coords logical_to_translated_coords(tt_logical_coords logical_coords);
+
+    tt_logical_coords physical_to_logical_coords(tt_physical_coords physical_coords);
+    tt_virtual_coords physical_to_virtual_coords(tt_physical_coords physical_coords);
+    tt_translated_coords physical_to_translated_coords(tt_physical_coords physical_coords);
+
+    tt_logical_coords virtual_to_logical_coords(tt_virtual_coords virtual_coords);
+    tt_physical_coords virtual_to_physical_coords(tt_virtual_coords virtual_coords);
+    tt_translated_coords virtual_to_translated_coords(tt_virtual_coords virtual_coords);
+
+    tt_logical_coords translated_to_logical_coords(tt_translated_coords translated_coords);
+    tt_physical_coords translated_to_physical_coords(tt_translated_coords translated_coords);
+    tt_virtual_coords translated_to_virtual_coords(tt_translated_coords translated_coords);
+
+    void perform_harvesting(std::size_t harvesting_mask);
+
+private:
+    CoordinateManager* coordinate_manager = nullptr;
```

Comment:
I don't see where this is `delete`'d, which is why I suggested `std::unique_ptr<CoordinateManager>`